app/models/daily_update.rb

```diff
@@ -0,0 +1,48 @@
+class DailyUpdate < ApplicationRecord
+  has_many :daily_update_units
+
+  class << self
+    def current
+      collection = self.order(year: :desc, month: :desc, day: :desc).limit(1)
+      collection.first
+    end
+
+    def queued_updates?
+      collection = self.where(proceeded: false)
+      !collection.empty?
+    end
+  end
+
+  def date
+    string_date = [year, month, day].join('/')
+    Date.parse(string_date)
+  end
+
+  def newer?(telltale_date)
+    self.date > telltale_date
+  end
+
+  def status
+    child_status = daily_update_units_status
+
+    if child_status.all? { |status| status == 'PENDING' }
+      return 'PENDING'
+
+    elsif child_status.all? { |status| status == 'COMPLETED' }
+      return 'COMPLETED'
+
+    elsif child_status.any? { |status| status == 'LOADING' }
+      return 'LOADING'
+
+    elsif child_status.any? { |status| status == 'ERROR' }
+      return 'ERROR'
+
+    elsif child_status.empty?
+      return 'PENDING'
```

Comment:
je mettrai autre chose que `PENDING` genre `NOT PROCEEDED` au niveau du code ça changera pas grand chose, mais pour la compréhension on verra bien qu'on distingue 2 cas différents.

En tout cas ça m'aurait aider pour le cas où `proceeded=false & 0 unit` != `proceeded=true & x unit pending`

Reply:
Je valide, je vais mettre `QUEUED` pour ce cas là